osmtm/views/task.py

```diff
@@ -107,21 +134,21 @@ def task_xhr(request):
     task_id = request.matchdict['task']
     project_id = request.matchdict['project']
 
-    filter = and_(TaskState.task_id == task_id,
-                  TaskState.project_id == project_id)
+    ancestors = get_task_ancestors(task_id, project_id)
+
+    filter = and_(and_(TaskState.task_id.in_(ancestors),
+                  TaskState.project_id == project_id),
+                  TaskState.state != TaskState.state_ready)
     states = DBSession.query(TaskState).filter(filter) \
         .order_by(TaskState.date).all()
-    # remove the first state (task creation with state==ready)
-    states.pop(0)
 
-    filter = and_(TaskLock.task_id == task_id,
-                  TaskLock.project_id == project_id)
+    filter = and_(TaskLock.task_id.in_(ancestors),
+                  TaskLock.project_id == project_id,
+                  TaskLock.lock is not None)
     locks = DBSession.query(TaskLock).filter(filter) \
         .order_by(TaskLock.date).all()
-    # remove the first lock (task creation)
-    locks.pop(0)
```

Comment:
On the dev instance, I am seeing 'unlocked' for all tiles at the time of their creation. I think this might need to be retained.